src/utils.js

```diff
@@ -672,6 +674,32 @@ function getCompileFn(implementation, options) {
       };
     }
 
+    if (options.api === "modern-compiler") {
+      return async (sassOptions) => {
+        // eslint-disable-next-line no-underscore-dangle
+        const webpackCompiler = loaderContext._compiler;
+        const { data, ...rest } = sassOptions;
+
+        // Some people can run the loader in a multi-threading way;
+        // there is no webpack compiler object in such case.
+        if (webpackCompiler) {
+          const key = isDartSass ? "dart-sass" : "sass-embedded";
+          if (!sassModernCompilers[key]) {
```

Comment:
I'm not sure if `getCompileFn()` can be invoked with different (implementation) options for a single instance of the sass-loader module, but I guess it can? In that case we should maintain a compiler instance for each `implementation`.

Reply:
I see, let's use `WeakCache` with options to prevent such behaviour